### PR TITLE
wip. more merge stuff. updated a bunch of gl conflicts.

### PR DESCRIPTION
**Pull request by cgaggar**

### CODE REVIEW

**Comment by themelissabrown** [comment]
add to whitelist

**Comment by AndrewMeadows** [comment]
Jenkins failed to build this PR on all platforms.  A subset of the windows compile errors are:
```
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(16): error C2039: 'bufferCount': is not a member of 'gpu::Backend'
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(16): error C2065: 'bufferCount': undeclared identifier
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(16): error C2228: left of '.decrement' must have class/struct/union
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(17): error C2039: 'bufferGPUMemSize': is not a member of 'gpu::Backend'
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(17): error C2065: 'bufferGPUMemSize': undeclared identifier 
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(17): error C2228: left of '.update' must have class/struct/union
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(32): error C2039: 'bufferCount': is not a member of 'gpu::Backend'
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(32): error C2065: 'bufferCount': undeclared identifier
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(32): error C2228: left of '.increment' must have class/struct/union
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(33): error C2039: 'bufferGPUMemSize': is not a member of 'gpu::Backend'
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(33): error C2065: 'bufferGPUMemSize': undeclared identifier 
libraries\gpu-gl\src\gpu\gl\GLBuffer.cpp(33): error C2228: left of '.update' must have class/struct/union
```

File: libraries/gpu-gl-android/src/gpu/gl/GLTexelFormat.cpp [hunk 1]
```diff
@@ -225,11 +225,13 @@ GLTexelFormat GLTexelFormat::evalGLTexelFormat(const Element& dstFormat, const E
...
-                qDebug() << "TODO: GLTexelFormat.cpp:evalGLTexelFormat GL_COMPRESSED_RED_RGTC1";
-                //texel.internalFormat = GL_COMPRESSED_RED_RGTC1;
-                break;
-
+				//CLIMAX_MERGE_START
+            // case gpu::COMPRESSED_R:
+                // qDebug() << "TODO: GLTexelFormat.cpp:evalGLTexelFormat GL_COMPRESSED_RED_RGTC1";
+                // //texel.internalFormat = GL_COMPRESSED_RED_RGTC1;
+                // break;
+				//CLIMAX_MERGE_END
```
Comment (AndrewMeadows) [inline libraries/gpu-gl-android/src/gpu/gl/GLTexelFormat.cpp]: Commented out code should be removed as cruft, unless there is a reason for keeping it around.  If there is a reason then it should be described in a comment.

File: libraries/gpu/src/gpu/Context.cpp [hunk 2]
```diff
@@ -242,82 +221,195 @@ Backend::TransformCamera Backend::TransformCamera::getEyeCamera(int eye, const S
...
+void Context::updateBufferGPUMemoryUsage(Size prevObjectSize, Size newObjectSize) {
+    if (prevObjectSize == newObjectSize) {
+        return;
+    }
+    if (newObjectSize > prevObjectSize) {
+        _bufferGPUMemoryUsage.fetch_add(newObjectSize - prevObjectSize);
+    } else {
+        _bufferGPUMemoryUsage.fetch_sub(prevObjectSize - newObjectSize);
+    }
+}
```
Comment (AndrewMeadows) [inline libraries/gpu/src/gpu/Context.cpp]: Maybe this should just be:
```
void Context::updateBufferGPUMemoryUsage(Size prevObjectSize, Size newObjectSize) {
    if (newObjectSize > prevObjectSize) {
        _bufferGPUMemoryUsage.fetch_add(newObjectSize - prevObjectSize);
    } else if (newObjectSize < prevObjectSize) {
        _bufferGPUMemoryUsage.fetch_sub(prevObjectSize - newObjectSize);
    }
}
```
Unless for some reason we expect the equality case to be the most common mode in which this is called: whatever the most common mode is... it should be the first check.

**Pull request merged**